### PR TITLE
multicluster: fix panic caused by invalid kubeconfig

### PR DESCRIPTION
**Pull request by hzxuzhonghu**
fixes: #13412

### CODE REVIEW

**Comment by hzxuzhonghu** [comment]
maybe this should also cherry-pick to 1.1 branch

File: pilot/pkg/config/clusterregistry/multicluster.go [hunk 1]
```diff
@@ -119,6 +119,10 @@ func (m *Multicluster) DeleteMemberCluster(clusterID string) error {
 	m.m.Lock()
 	defer m.m.Unlock()
 	m.serviceController.DeleteRegistry(clusterID)
+	if _, ok := m.remoteKubeControllers[clusterID]; !ok {
+		log.Infof("cluster %s not exist, maybe caused by invalid kubeconfig", clusterID)
```
Comment (john-a-joyce) [inline pilot/pkg/config/clusterregistry/multicluster.go]: minor language Nit.   Could you change this to "cluster %s does not exist"

Comment (hzxuzhonghu) [reply pilot/pkg/config/clusterregistry/multicluster.go]: ok

**Comment by hzxuzhonghu** [comment]
/assign @howardjohn 
Maybe should also to 1.1

File: pilot/pkg/config/clusterregistry/multicluster.go [hunk 2]
```diff
@@ -119,6 +119,10 @@ func (m *Multicluster) DeleteMemberCluster(clusterID string) error {
 	m.m.Lock()
 	defer m.m.Unlock()
 	m.serviceController.DeleteRegistry(clusterID)
+	if _, ok := m.remoteKubeControllers[clusterID]; !ok {
+		log.Infof("cluster %s does not exist, maybe caused by invalid kubeconfig", clusterID)
+		return nil
+	}
 	close(m.remoteKubeControllers[clusterID].stopCh)
 	delete(m.remoteKubeControllers, clusterID)
 	if m.XDSUpdater != nil {
```
Comment (howardjohn) [inline pilot/pkg/config/clusterregistry/multicluster.go]: Do so still need this cleanup?

Comment (hzxuzhonghu) [reply pilot/pkg/config/clusterregistry/multicluster.go]: no, actually this is a invalid cluster defined in secret, but has not been initiated successfully.
So no need to cleanup.

**Pull request merged**